### ISSUE TITLE
parse failure on <a> after <svg>

### ISSUE THREAD

**Issue by julfy** [aantron/markup.ml#33]
minimal repro:
```
Markup.("<svg><textPath></textPath></svg> <a></a>" |> string |> parse_html |> signals |> write_html |> to_string);;
```
it fails only when tags subject to `Foreign.adjust_svg_tag_name` are used, but i don't see a reason why it should be the case
also, from what i gathered, it fails only when `<a>` is on the same nesting level or directly higher than `<svg>`
eg. these still fail:
```
Markup.("
<div><svg><textPath></textPath></svg></div> 
<a></a>" |> string |> parse_html |> signals |> write_html |> to_string);;

Markup.("
<svg><altglyph></altglyph></svg> 
<div></div> 
<a></a>" |> string |> parse_html |> signals |> write_html |> to_string);;
```
these don't:
```
Markup.("
<div><svg><textPath></textPath></svg></div> 
<div><a></a></div>" |> string |> parse_html |> signals |> write_html |> to_string);;

Markup.("
<svg><circle></circle></svg>
<a></a>" |> string |> parse_html |> signals |> write_html |> to_string);;

Markup.("
<svg><textPath></textPath></svg> 
<div><a></a></div>" |> string |> parse_html |> signals |> write_html |> to_string);;
```

**Comment by aantron** [comment]
Thanks for the report and the example. I was able to reproduce it, and will try to fix it over the weekend!

**Comment by aantron** [comment]
The attached commit should fix the problem, let me know if it works. If it does, let me know if you'd like a speedy release of Markup.ml to opam :)

The problem was happening because the HTML tokenizer [emits all tag names in lowercase](https://www.w3.org/TR/html52/syntax.html#tokenizer-tag-name-state):

> Uppercase ASCII letter 
>
> Append the lowercase version of the current input character (add 0x0020 to the character’s code point) to the current tag token’s tag name.

So, `<textPath>` is tokenized as `<textpath>`, and `</textPath>` is tokenized as `</textpath>`.

After that, when the HTML parser is parsing SVG, it [converts `<textpath>` "back" to `<textPath>`](https://www.w3.org/TR/html52/syntax.html#the-rules-for-parsing-tokens-in-foreign-content). This is done by the `Foreign.adjust_svg_tag_name` function, which you found.

The problem was that Markup.ml was pushing this "fixed" `<textPath>` onto its parsing stack, but then looking for `</textpath>` (all lowercase) when it came time to pop the element.

The solution is to push the original, lowercased tokens (like `<textpath>`) onto the stack, so that they are found and popped by the corresponding lowercased end tag tokens.

I added a test case to the commit to ensure no regressions. It's based on your repro case, so thanks again for that!

**Comment by julfy** [comment]
Thank you for such a swift fix! Works fine on real data. 
Release would surely be convenient, thank you :)

**Comment by aantron** [comment]
Great, thanks.

I'll take a look at a couple other issues in the next few days, then do a release. Will post here when it's in opam :)

**Comment by aantron** [comment]
This is now released and installable from opam as `markup` 0.7.7.

**Comment by julfy** [comment]
:clap:
